src/bloqade/ir/control/field.py

```diff
@@ -209,9 +218,39 @@ class Field(FieldExpr):
 
     drives: Dict[SpatialModulation, Waveform]
 
-    def __hash__(self) -> int:
+    @cached_property
+    def _hash_value(self):
         return hash(frozenset(self.drives.items())) ^ hash(self.__class__)
 
+    def __hash__(self) -> int:
+        return self._hash_value
+
+    def canoncialize(self) -> "Field":
```

Comment:
put a line of doc here to explain what kind of canonicalization is done here? the spell is wrong btw